src/contracts/core/RewardsCoordinator.sol

```diff
@@ -143,6 +143,43 @@ contract RewardsCoordinator is
         }
     }
 
+    /**
+     * @notice Exact same behavior as createAVSRewardsSubmission but allows a rewardsForAllSubmitter to
+     * create a rewards submission on behalf of an AVS. Stakers delegated to operators who are registered to the `avs`
+     * will receive these rewards.
+     * @param avs The address of the AVS for which the rewards submission is being made
+     * @param rewardsSubmissions The rewards submissions being created
+     * @dev Expected to be called by a rewardsForAllSubmitter
+     * @dev The duration of the `rewardsSubmission` cannot exceed `MAX_REWARDS_DURATION`
+     * @dev The tokens are sent to the `RewardsCoordinator` contract
+     * @dev Strategies must be in ascending order of addresses to check for duplicates
+     * @dev This function will revert if the `rewardsSubmission` is malformed,
+     * e.g. if the `strategies` and `weights` arrays are of non-equal lengths
+     */
+    function createRewardsSubmissionForAVS(
+        address avs,
+        RewardsSubmission[] calldata rewardsSubmissions
+    )
+        external
+        onlyWhenNotPaused(PAUSED_AVS_REWARDS_SUBMISSION)
+        onlyRewardsForAllSubmitter
+        nonReentrant
+    {
+        for (uint256 i = 0; i < rewardsSubmissions.length; i++) {
+            RewardsSubmission calldata rewardsSubmission = rewardsSubmissions[i];
+            uint256 nonce = submissionNonce[avs];
+            bytes32 rewardsSubmissionHash = keccak256(abi.encode(avs, nonce, rewardsSubmission));
+
+            _validateRewardsSubmission(rewardsSubmission);
+
+            isAVSRewardsSubmissionHash[avs][rewardsSubmissionHash] = true;
+            submissionNonce[avs] = nonce + 1;
+
+            emit AVSRewardsSubmissionCreated(avs, nonce, rewardsSubmissionHash, rewardsSubmission);
```

Comment:
Note that we cannot distinguish, via events off-chain, between rewards submitted _by_ AVSs and _on behalf_ of AVSs. Is this by design? If we do care, then the bronze/silver tables need to be updated.